src/fillTank.test.js

```diff
@@ -1,11 +1,94 @@
 'use strict';
 
+const { fillTank } = require('./fillTank');
+
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  test('should fill the tank only up to its capacity even if'
+    + 'more money is available', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40, fuelRemains: 38,
+      },
+    };
+    const fuelPrice = 5;
+
+    fillTank(customer, fuelPrice);
+    expect(customer.vehicle.fuelRemains).toBe(40);
+    expect(customer.money).toBe(2990);
+  });
+
+  test('should round down correctly with floating-point'
+   + 'precision issues', () => {
+    const customer = {
+      money: 0.9999,
+      vehicle: {
+        maxTankCapacity: 40, fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 0.33333;
 
-  it('should ', () => {
+    fillTank(customer, fuelPrice);
+    expect(customer.vehicle.fuelRemains).toBe(10.9);
 
+    expect(customer.money).toBe(0);
   });
 
-  // write tests here
+  test('should pour as much as affordable', () => {
+    const customer = {
+      money: 50,
+      vehicle: {
+        maxTankCapacity: 40, fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 10;
+
+    fillTank(customer, fuelPrice);
+    expect(customer.vehicle.fuelRemains).toBe(13);
+    expect(customer.money).toBe(0);
+  });
+
+  test('should fill the specified amount', () => {
+    const customer = {
+      money: 500,
+      vehicle: {
+        maxTankCapacity: 40, fuelRemains: 8,
+      },
+    };
+    const fuelPrice = 5;
+    const amount = 20;
+
+    fillTank(customer, fuelPrice, amount);
+    expect(customer.vehicle.fuelRemains).toBe(28);
+    expect(customer.money).toBe(400);
+  });
+
+  test('should not pour if amount is less than 2 liters', () => {
```

Comment:
The code does not show the logic for determining the fuel filling conditions. To properly test this condition, the fillTank function should contain logic that prevents fueling in such cases.